Content.Client/Instruments/UI/ChannelsMenu.xaml.cs

```diff
@@ -51,15 +53,23 @@ private void OnClearPressed(BaseButton.ButtonEventArgs obj)
         }
     }
 
-    public void Populate(InstrumentComponent? instrument)
+    public void Populate(InstrumentComponent? instrument, ActiveInstrumentComponent? activeInstrument)
     {
         ChannelList.Clear();
 
         for (int i = 0; i < RobustMidiEvent.MaxChannels; i++)
         {
-            var item = ChannelList.AddItem(_owner.Loc.GetString("instrument-component-channel-name",
-                ("number", i)), null, true, i);
+            var label = _owner.Loc.GetString("instrument-component-channel-name",
+                ("number", i));
+            if (activeInstrument != null
+                && activeInstrument.Channels.TryGetValue(i, out var resolvedMidiChannel)
+                && resolvedMidiChannel != null
+                )
+            {
+                label = resolvedMidiChannel;
```

Comment:
I would keep the numbers of the MIDI channel in there just for clarity. There's definitely MIDIs with multiple tracks with the same instrument, so some distinction is good.